backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -16,14 +21,70 @@ import UserService from "../../services/implementations/userService";
 import IEmailService from "../../services/interfaces/emailService";
 import IUserService from "../../services/interfaces/userService";
 import { Role } from "../../types";
-import { CourseVisibilityAttributes } from "../../models/course.model";
+import { CourseVisibilityAttributes, Module } from "../../models/course.model";
 import { assertNever } from "../../utilities/errorUtils";
 
 const courseService: ICourseService = new CourseService();
 const userService: IUserService = new UserService();
 const emailService: IEmailService = new EmailService(nodemailerConfig);
 const authService: IAuthService = new AuthService(userService, emailService);
 
+function serializeToModuleDTO(modules: { [id: string]: Module }): ModuleDTO[] {
+  return Object.entries(modules).map(([key, val]) => ({ id: key, ...val }));
+}
+
+function deserializeModuleDTO(
+  moduleDTOs: ModuleDTO[],
+): { [id: string]: Module } {
+  const modules: { [id: string]: Module } = {} as { [id: string]: Module };
+
+  moduleDTOs.forEach((moduleDTO) => {
+    const moduleObj = {
+      title: moduleDTO.title,
+      description: moduleDTO.description,
+      image: moduleDTO.image,
+      previewImage: moduleDTO.previewImage,
+      published: moduleDTO.published,
+      lessons: moduleDTO.lessons,
+    };
+
+    // If no UUID is defined, create one
+    modules[moduleDTO?.id ?? uuidv4()] = moduleObj;
```

Comment:
When do we expect the `moduleDTO` to have an ID vs. not have an ID? If it should always have an ID, we should expect to always have an ID and handle this more forcefully rather than absorbing the error (whether this is showing an error, or making a mutation in the db to fix)

Reply:
Module DTO might not have an ID when we have a request to create a new course with a new module, in which case we create the ID here, but I agree that doing this more forcefully elsewhere might be cleaner. The other option is generating an ID in the client but that's even more anti-pattern.